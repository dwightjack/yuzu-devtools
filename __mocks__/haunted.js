const haunted = require('haunted/index');

module.exports = {
  ...haunted,
  useContext: jest.fn(),
};
