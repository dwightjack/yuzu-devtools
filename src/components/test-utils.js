export const mockComponent = (tag) => (props) => {
  const el = document.createElement(tag);
  Object.keys(props).forEach((key) => {
    el.setAttribute(key, props[key]);
  });
  return el;
};
