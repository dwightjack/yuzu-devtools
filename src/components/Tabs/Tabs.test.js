import { virtual } from 'haunted';
import { fireEvent } from 'dom-testing-library';
import { toHTMLAsync, cycle } from '../test-utils';

const Tabs = virtual(require('./Tabs').default);

const tabs = [
  { id: 'first', label: 'FIRST' },
  { id: 'second', label: 'SECOND' },
];

describe('Tabs', () => {
  test('default snapshot', async () => {
    const { html } = await toHTMLAsync(Tabs, { tabs });
    expect(html).toMatchSnapshot();
  });

  test('renders two slots', async () => {
    const wrapper = await toHTMLAsync(Tabs, { tabs });
    tabs.forEach(({ id }) => {
      expect(wrapper.find(`slot[name="${id}"]`)).not.toBeUndefined();
    });
  });

  test('first tab is selected by default', async () => {
    const wrapper = await toHTMLAsync(Tabs, { tabs });
    const first = wrapper.find('[role="tabpanel"]');
    expect(wrapper.find(`[role="tabpanel"]:not(hidden)`)).toBe(first);
  });

  test('selects the second tab', async () => {
    const wrapper = await toHTMLAsync(Tabs, { tabs });
    const tab = wrapper.find('a#second-tab');
    const second = wrapper.find('#second-panel');
    fireEvent.click(tab);
    await cycle();
    expect(wrapper.find(`[role="tabpanel"]:not([hidden])`).id).toBe(second.id);
  });
});
