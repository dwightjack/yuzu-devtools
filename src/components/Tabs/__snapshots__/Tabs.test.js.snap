// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs default snapshot 1`] = `
<ul role="tablist">
  <li role="presentation">
    <a role="tab"
       href="#first-panel"
       id="first-tab"
       aria-selected="true"
    >
      FIRST
    </a>
  </li>
  <li role="presentation">
    <a role="tab"
       href="#second-panel"
       id="second-tab"
       aria-selected="false"
    >
      SECOND
    </a>
  </li>
</ul>
<section role="tabpanel"
         id="first-panel"
         aria-labelledby="first-tab"
>
  <slot name="first">
  </slot>
</section>
<section role="tabpanel"
         id="second-panel"
         aria-labelledby="second-tab"
         hidden
>
  <slot name="second">
  </slot>
</section>
`;
